### DISCUSSION TITLE
sha256 or sha of the gem in the versions API and versions tables of pg-dump

### DISCUSSION THREAD

**Discussion by nareshdevasani** [ruby/rubygems#6433] [Q&A] [answered]
Hello, I need help in understanding below data: How could I transform the dump's `sha256` value to `sha` of the API?

API: https://rubygems.org/api/v2/rubygems/catsay/versions/0.2.1.json
sha: e4cdf7fb3a7666764d7c3b06531bf388c7597ef5a7cf04ef54b816f0122a1b40

pg-dump of rubygems has below data for the same gem and version:

id: 634090
number: 0.2.1
rubygem_id: 95809
     full_name: catsay-0.2.1
`        sha256: 5M33+zp2ZnZNfDsGUxvziMdZfvWnzwTvVLgW8BIqG0A= `
         info_checksum: 1848f2d9d098df0129209f628c0b0115

Thanks in advance, any help is appreciated.

Regards,
Naresh

**Comment by composerinteralia** [comment, marked as answer]
Looks like:

```rb
"5M33+zp2ZnZNfDsGUxvziMdZfvWnzwTvVLgW8BIqG0A=".unpack1("m0").unpack1("H*")
#=> "e4cdf7fb3a7666764d7c3b06531bf388c7597ef5a7cf04ef54b816f0122a1b40"
```
from https://github.com/rubygems/rubygems.org/blob/4985a916480e04e9876d3e76ccdc87cb5beabe9b/app/models/version.rb#L359

**Reply by nareshdevasani** [reply]
this helps a lot. Thank you very much @composerinteralia
